### PR TITLE
feature(wallet-mobile): Dapp explorer UI

### PR DESCRIPTION
**Pull request by stackchain**

### CODE REVIEW

File: apps/wallet-mobile/src/WalletNavigator.tsx [hunk 1]
```diff
@@ -21,25 +22,22 @@ import {
...
-import {isHaskellShelley} from './yoroi-wallets/cardano/utils'
 import {useHasWallets} from './yoroi-wallets/hooks'
 
 const Tab = createBottomTabNavigator<WalletTabRoutes>()
 const WalletTabNavigator = () => {
   const strings = useStrings()
   const {colors} = useStyles()
-  const wallet = useSelectedWallet()
-  const initialRoute = isHaskellShelley(wallet.walletImplementationId) ? 'staking-dashboard' : 'history'
+  const initialRoute = /* isHaskellShelley(wallet.walletImplementationId) ? 'staking-dashboard' :*/ 'history'
```
Comment (stackchain) [inline apps/wallet-mobile/src/WalletNavigator.tsx:40]: this can be dropped

File: apps/wallet-mobile/src/WalletNavigator.tsx [hunk 2]
```diff
@@ -117,7 +115,7 @@ const WalletTabNavigator = () => {
           )}
         </Tab.Screen>
 
-        {isHaskellShelley(wallet.walletImplementationId) && (
+        {/* {isHaskellShelley(wallet.walletImplementationId) && (
```
Comment (stackchain) [inline apps/wallet-mobile/src/WalletNavigator.tsx:118]: check the feature flag

File: apps/wallet-mobile/src/assets/img/dApp/ada-handle.png
Comment (stackchain) [inline apps/wallet-mobile/src/assets/img/dApp/ada-handle.png:1]: I'm guessing these icons are mocked for now, later they will come from the configuration manageR?

File: apps/wallet-mobile/src/components/Icon/Backward.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
```
Comment (stackchain) [inline apps/wallet-mobile/src/components/Icon/Backward.tsx:1]: import * React... for all screens

File: apps/wallet-mobile/src/components/Icon/Backward.tsx
Comment (stackchain) [inline apps/wallet-mobile/src/components/Icon/Backward.tsx:1]: @michaeljscript I don't see the stories

File: apps/wallet-mobile/src/assets/img/dApp/ada-handle.png
Comment (michalszorad) [reply apps/wallet-mobile/src/assets/img/dApp/ada-handle.png:1]: Yes

File: apps/wallet-mobile/src/features/Discover/BrowserNavigator.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import {BrowseDappScreen} from './useCases/BrowseDappScreen/BrowseDappScreen'
+import {BrowseSearchDappScreen} from './useCases/BrowseSearchDappScreen/BrowseSearchDappScreen'
+
+const Tab = createStackNavigator<BrowserRoutes>()
+
+export const BrowserNavigator = () => {
+  const {styles} = useStyles()
+
+  return (
+    <SafeAreaView edges={['left', 'right', 'top']} style={styles.root}>
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Discover/BrowserNavigator.tsx:17]: if it's a screen NavigateDappScreen

File: apps/wallet-mobile/src/features/Discover/common/BrowserProvider.tsx [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+  const storage = useAsyncStorage()
+  const wallet = useSelectedWallet()
+  const browserStorage = storage.join(`wallet/${wallet.id}/${storageRootBrowser}/`)
+
+  const [browserState, dispatch] = React.useReducer(browserReducer, {
+    ...initialBrowserState,
+    ...initialState,
+  })
+
+  useEffect(() => {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Discover/common/BrowserProvider.tsx:64]: React.useEffect

**Comment by stackchain** [comment]
@michaeljscript looks solid, basically some adjust to import react, and consume react hooks with React.use<blabla> bringing it in, please adjust the feature flag.

**Pull request merged**